libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/idp/IDPInititatedLoginReceiver.java

```diff
@@ -57,18 +62,41 @@ public void onReceive(Context context, Intent intent) {
                 final Bundle extras = intent.getExtras();
                 if (extras != null) {
                     userHint = extras.getString(USER_HINT_KEY);
+                    spActivityName = extras.getString(SP_ACTVITY_NAME_KEY);
                 }
 
                 // Launches login flow if the user doesn't already exist on the SP app.
                 if (!doesUserExist()) {
                     launchLoginActivity();
+                } else {
+
+                    // If user hint was passed in, switches to the specified user.
+                    UserAccountManager.getInstance().switchToUser(getUserForHint());
```

Comment:
If the user already exists on the SP app, we should switch to that user before launching whatever component the IDP passed in.